src/Serializer/JsonApiSerializer.php

```diff
@@ -84,4 +124,176 @@ public function sideloadIncludes()
     {
         return true;
     }
+
+    public function injectData($data, $includedData)
+    {
+        $relationships = $this->parseRelationships($includedData);
+
+        if (!empty($relationships)) {
+            $data = $this->fillRelationships($data, $relationships);
+        }
+
+        return $data;
+    }
+
+    /**
+     * The JSON API specification does not allow the root object to be included
```

Comment:
Your short description looks like this:

```
The JSON API specification does not allow the root object to be included
```

and your long description:

```
into the sideloaded `included`-array. We have to make sure it is
filtered out, in case some object links to the root object in a
relationship.
```